Common/Exceptions/ExceptionParser.cs

```diff
@@ -43,13 +43,20 @@ public ExceptionParser()
         /// <returns>Parsed exception</returns>
         public Exception Parse(Exception exception)
         {
-            if (exception is PythonException || exception.InnerException is PythonException)
+            try
             {
-                return _pythonUserExceptionParser.Parse(exception);
+                if (exception is PythonException || exception.InnerException is PythonException)
+                {
+                    return _pythonUserExceptionParser.Parse(exception);
+                }
+                else
+                {
+                    return _cSharpUserExceptionParser.Parse(exception);
+                }
             }
-            else
+            catch
             {
-                return _cSharpUserExceptionParser.Parse(exception);
+                return exception;
```

Comment:
Let's at least log the exception that was thrown here. This will help us find/debug errors in the exception parsers